gui/matRad_Widget.m

```diff
@@ -137,9 +147,34 @@ function enableWindowCallback(this,enable)
             end
         end
     end
+
+    methods (Sealed)
+        
+        %Update function to be called when the widget should be updated
+        %(i.e. by listener to the workspaceChanged event or in
+        %initialization)
+        %Implement the protected doUpdate function in a subclass to define
+        %the update process
+        function update(this,evt)
+            try
+                if nargin == 2
+                    this.doUpdate(evt);
+                else
+                    this.doUpdate();
+                end
+            catch ME
+                %Error Handling
+                this.showWarning('Update failed',ME); %We only show a warning to not halt execution.
+            end
+        end
+    end
     
     methods (Access = protected)
         
+        function this = doUpdate(this)
```

Comment:
Should probably be doUpdate(this,~) to be consistent in number of accepted arguments?